sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```
@@ -0,0 +1,1058 @@
+          if (recordReader != null) {
+            recordReader.initialize(split.getSplit(), taskAttemptContext);
+            if (recordReader.nextKeyValue()) {
+              recordsReturned++;
+              return true;
+            }
+          } else {
+            throw new IOException(String.format(
+                HadoopInputFormatIOConstants.NULL_CREATE_RECORDREADER_ERROR_MSG,
+                inputFormatObj.getClass()));
+          }
+          recordReader = null;
+        } catch (InterruptedException e) {
+          throw new IOException(
+              "Could not read because the thread got interrupted while "
+              + "reading the records with an exception: ",
+              e);
+        }
+        doneReading = true;
+        return false;
+      }
+
+      @Override
+      public boolean advance() throws IOException {
+        try {
+          progressValue = new AtomicDouble(getProgress());
+          if (recordReader != null && recordReader.nextKeyValue()) {
+            recordsReturned++;
+            return true;
+          }
+          doneReading = true;
+        } catch (InterruptedException e) {
+          throw new IOException("Unable to read data: ", e);
+        }
+        return false;
+      }
+
+      @Override
+      public KV<K, V> getCurrent() throws NoSuchElementException {
+        K key = null;
+        V value = null;
+        try {
+          // Transform key if translation function is provided.
+          key =
+              transformKeyOrValue((T1) recordReader.getCurrentKey(), keyTranslationFunction,
+                  keyCoder);
+          // Transform value if if translation function is provided.
+          value =
+              transformKeyOrValue((T2) recordReader.getCurrentValue(), valueTranslationFunction,
+                  valueCoder);
+        } catch (IOException | InterruptedException e) {
+          LOG.error(HadoopInputFormatIOConstants.GET_CURRENT_ERROR_MSG + "{}", e);
```

Here, {} is the placeholder for exception e used for String concatenation.
We are now not suppressing the exception, instead we are throwing an "IllegalStateException" to avoid silent data loss scenarios.